client/src/entrypoints/contrib/typed_table_block/typed_table_block.js

```diff
@@ -235,6 +235,9 @@ export class TypedTableBlock {
     column.headingInput.name = this.prefix + '-column-' + column.id + '-heading';
     column.headingInput.className = 'column-heading';
     column.headingInput.placeholder = this.blockDef.meta.strings.COLUMN_HEADING;
+    column.headingInput.setAttribute(
+      'aria-label', `${h(this.blockDef.meta.strings.COLUMN_HEADING)} for ${h(blockDef.meta.label)} column`
```

Comment:
The `COLUMN_HEADING` string (and potentially blockDef.meta.label) is translated, so for people using other languages, this is likely to end up as a mix of English and their own language. This would be fixable by adding 'column heading for X' as [another translatable string](https://github.com/wagtail/wagtail/blob/24b99a5ebc4b25986b89bbbfa0de6e93acfe6dcc/wagtail/contrib/typed_table_block/blocks.py#L286) - we don't currently have any provision for placeholders in those strings, but it would be easy enough to implement one with a simple string search and replace.

I'm wondering if `blockDef.meta.label` is the most useful thing to use to identify the column, though, given that it's typically going to be something generic like "rich text column"? I'm assuming we can't just say "column heading" because we're expected to make them unique, but adding the column type doesn't guarantee that either. Maybe we can use the column number instead - "column heading for column 1"?

Reply:
Yes, I think further work is needed here. Even without the `aria-label`, the field is already announced as "Column heading" thanks to its placeholder, so it’s not so bad. Having the column number would indeed be nice so it’s possible to distinguish between them when navigating the page linearly.

I’d recommend running a few options past @wagtail/accessibility, as it doesn’t feel clear to me which labels would be most helpful.

Reply:
Placeholders are really not prefered as the text label to be used by screen readers since it is not a standard, so preferably a `<label>` tag with some class to hide it visually if needed, or at least an aria-label is definitely needed. PS: https://www.htmhell.dev/24-a-placeholder-is-not-a-label/ , https://www.smashingmagazine.com/2018/06/placeholder-attribute/ , https://codepen.io/stevef/post/placeholder-the-piss-take-label

Reply:
@SaptakS by "which labels would be most helpful", I meant the textual content rather than mean of implementation. In a table like this it feels like each cell field should announce its row/column number (or "column 1 heading" in the case of the `th`), but this could get pretty verbose.

The placeholder is taken into account in the accessible name computation (see [HTML-AAM](https://www.w3.org/TR/html-aam-1.0/#accessible-name-and-description-computation)) as the last resort, hence why I think this can wait until we have the best label possible. I still wouldn’t recommend placeholder over `aria-label`, but at least there is _something_ announced currently.

Reply:
Agree with everything you said. Just wanted to note that we should finally at some point replace placeholder with proper label, since many screen readers don't respect placeholders.